python/paddle/fluid/trainer.py

```diff
@@ -301,7 +313,8 @@ def _train_by_any_executor(self, event_handler, exe, num_epochs, reader):
                 begin_event = BeginStepEvent(epoch_id, step_id)
                 event_handler(begin_event)
                 if begin_event.fetch_metrics:
-                    metrics = exe.run(feed=data,
+                    metrics = exe.run(feed=data_feed_handler(data)
```

Comment:
@jacquesqiao : I think probably this is incorrect right? We would need data from the reader, rather than the `feeder.decorate_reader` right?

Reply:
So do you have any recommendation on how we could handle this data_feed_handler with the `decorate_reader`?